### DISCUSSION TITLE
Weird behavior when HLS muxer can't keep up

### DISCUSSION THREAD

**Discussion by Curid** [bluenviron/mediamtx#845] [Questions] [answered]
The input is a RTMP camera. HLS is set to always remux. Linux arm64  [1c6e20b](https://github.com/aler9/rtsp-simple-server/commit/1c6e20b274ab3fcc9b96c1ded777320fbb9e19b0)

The sequence number is incrementing normally for a few thousand requests then it randomly jumps 100+ and throws a bunch of `invalid FU-A packet (non-starting)`

```
prevSeq: 1001 seq: 1499                                                         
prevSeq: 1505 seq: 1009                                                         
prevSeq: 1009 seq: 1506                                                         
prevSeq: 1619 seq: 2127
prevSeq: 2324 seq: 2817
prevSeq: 2992 seq: 3474
prevSeq: 3491 seq: 3999
prevSeq: 4000 seq: 3494
prevSeq: 3497 seq: 4005
prevSeq: 4189 seq: 4691
```

**Comment by Curid** [comment, marked as answer]
Increasing the readBufferCount fixes it. Though I still wonder why it behaves like this on dropped packets.

**Comment by aler9** [comment]
The server uses a ring buffer to read packets from publishers and write packets to readers:
https://en.wikipedia.org/wiki/Circular_buffer

If the buffer fills before packets are written to readers, packets in the buffer are overridden.
readBufferCount controls the buffer size. By default it is set to a low value and must be tuned manually to avoid RAM exhaustion.

**Reply by Curid** [reply]
Oh now I understand. The writer temporarily gets ahead of the reader by 2 packets resulting in the sequence number jumping from 3491 to 3999-4000 and then back to 3494.

[a comment by an automated account is not shown]
